examples/Scikit.ipynb

```diff
@@ -0,0 +1,330 @@
+    "print('Example run in %.3f s' % run_time)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52a034e3",
+   "metadata": {},
+   "source": [
+    "## The \"Application\"\n",
+    "\n",
+    "We can use the `xaitk-saliency` high-level API to generate saliency maps that compare to the visualization above.\n",
+    "\n",
+    "Our \"application\" will accept a set of images, a blackbox image classifier, and a saliency generator and will generate saliency maps for each image provided.\n",
+    "It will then combine the heatmaps of each corresponding class and plot them.\n",
+    "This should give us a view of the decision boundaries for each class similar to the one above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "08fb1a3e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from smqtk_classifier import ClassifyImage\n",
+    "from xaitk_saliency import GenerateImageClassifierBlackboxSaliency\n",
+    "from functools import reduce\n",
+    "from math import ceil\n",
+    "\n",
+    "def app(\n",
+    "    images: np.ndarray,\n",
+    "    image_classifier: ClassifyImage,\n",
+    "    saliency_generator: GenerateImageClassifierBlackboxSaliency\n",
+    "):\n",
+    "    num_imgs = images.shape[0]\n",
+    "    \n",
+    "    # Generate saliency maps\n",
+    "    sal_maps = []\n",
+    "    for i in range(num_imgs):    \n",
+    "        ref_image = images[i].reshape(28, 28)\n",
+    "        sal_maps.append(saliency_generator(ref_image, image_classifier))\n",
+    "    \n",
+    "    # Combine respective class maps\n",
+    "    combined_maps = reduce(np.add, sal_maps)\n",
+    "    \n",
+    "    # Plot combined maps\n",
+    "    plt.figure(figsize=(10, 5))\n",
+    "    for i in range(num_imgs):\n",
+    "        max_val = combined_maps[i].max()\n",
+    "        plt.subplot(2, ceil(num_imgs/2), i + 1)\n",
+    "        plt.imshow(combined_maps[i], cmap=plt.cm.RdBu, vmin=-max_val, vmax=max_val)\n",
+    "        plt.axis('off')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e6d18472",
+   "metadata": {},
+   "source": [
+    "## Blackbox Classifier\n",
+    "\n",
+    "Here we wrap our logistic regression in a `ClassifyImage` class to comply with the API's interface."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "e7277f40",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MNISTClassifier (ClassifyImage):\n",
+    "    def get_labels(self):\n",
+    "        return [i for i in range(10)]\n",
+    "    \n",
+    "    def classify_images(self, image_iter):\n",
+    "        \n",
+    "        # Convert image set back to column vector form\n",
+    "        images = np.array([]).reshape(0, 28*28)\n",
+    "        for img in image_iter:\n",
+    "            images = np.vstack((images, img.reshape(1, -1)))\n",
+    "        \n",
+    "        return clf.decision_function(images)\n",
```

Comment:
This technically does not fit the output data type of the interface API (returns iterator of dict instances). This seems to happen to work due to the classes being integers and the output elements index-able arrays. Maybe we don't need to change the code, but at least comment that the API is not being "met" due to special circumstances.

Reply:
I think I fixed this? No totally sure.

Reply:
Partially: it's now set to return an iterator, however the parent API states an iterator of dictionaries. Also, images are documented to be input as 2- or 3-dim. A more "correct" implementation might be:
```python
class MNISTClassifier (ClassifyImage):
    ...

    def classify_images(self, image_iter):
        # Yes, "images" in this example case are really 1-dim (28*28=784).
        # MLP input needs a (n_samples, n_features) matrix input.
        images = np.asarray(list(image_iter))  # may fail because input is not consistent in shape
        images = images.reshape(-1, 28*28)  # may fail because input was not the correct shape
        return (
            {idx: p_i for idx, p_i in zip(range(10), p)}
            for p in mlp.predict_proba(images)
        )
```